### PR TITLE
add deposit contract wrapper

### PR DESCRIPTION
**Pull request by daiki44**
- [x] Deposit Contract Wrapper
- [x] fix Event Watcher

### CODE REVIEW

File: packages/contract/src/DepositContract.ts [hunk 1]
```diff
@@ -1,28 +1,244 @@
...
+                        {
+                          prim: 'Pair',
+                          args: [
+                            [
+                              {
+                                prim: 'Elt',
+                                args: [
+                                  { int: '0' },
+                                  {
+                                    bytes: `'${initialState.inputs[0].toHexString()}'`
```
Comment (syuhei176) [inline packages/contract/src/DepositContract.ts]: maybe we need to remove prefix `0x`.

File: packages/contract/src/DepositContract.ts [hunk 2]
```diff
@@ -1,28 +1,244 @@
...
+      //         prim: 'Pair',
+      //         args: [
+      //           { string: 'PREDICATE_ADDRESS' },
+      //           [{ bytes: '' }, { bytes: '' }]
+      //         ]
+      //       }
+      //     ]
+      //   }
+      // ]
+      const checkpointId = log.values[1].string
```
Comment (syuhei176) [inline packages/contract/src/DepositContract.ts]: `log.values[1].bytes` ?

File: packages/contract/src/events/TzEventWatcher.ts [hunk 3]
```diff
@@ -116,10 +132,16 @@ export default class EventWatcher implements IEventWatcher {
         if (handler) {
           const args = e.args[1] as MichelinePrim[]
           args.forEach(arg => {
+            // note: wanna change to MichelineBytes[]
+            const values = ((arg as MichelinePrim).args[1] as any[]).map(
+              b =>
+                JSON.parse(
+                  TezosLanguageUtil.hexToMicheline(b.bytes).code
+                ) as MichelinePrimItem
```
Comment (syuhei176) [inline packages/contract/src/events/TzEventWatcher.ts]: It seems good! maybe ;;)

**Comment by daiki44** [comment]
fixed! could you review this again? @syuhei176

**Pull request merged**